### PR TITLE
Fix refreshing targeted vm after snapshot removal

### PR DESCRIPTION
**Pull request by borod108**
Avoid refreshing the vm a second time if snapshot was removed.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1571291

depends on https://github.com/ManageIQ/manageiq/pull/17462

### CODE REVIEW

**Comment by borod108** [comment]
@pkliczewski can you please review?

**Comment by JPrause** [comment]
@borod108 if this PR can be backported, can you add the gaprindashvili/yes label.

File: app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb [hunk 1]
```diff
@@ -92,20 +92,16 @@ def hosts
...
-
-    manager.with_provider_connection(VERSION_HASH) do |connection|
-      select_vms(references(:vms)).each do |ems_ref|
+    @vms = manager.with_provider_connection(VERSION_HASH) do |connection|
+      select_vms(references(:vms)).map do |ems_ref|
         begin
-          v << connection.system_service.vms_service.vm_service(uuid_from_ems_ref(ems_ref)).get
+          connection.system_service.vms_service.vm_service(uuid_from_ems_ref(ems_ref)).get
         rescue OvirtSDK4::Error # when 404
           nil
```
Comment (masayag) [inline app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb]: if this potentially returns nil, shouldn't you add .compact to the result of the map ?

Comment (masayag) [inline app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb]: if this potentially returns nil, shouldn't you add .compact to the result of the map ?

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 5643660ddede56d74d9b1553e5b33b6611fb2cad
Author: Adam Grare <agrare@redhat.com>
Date:   Tue May 22 08:50:38 2018 -0400

    Merge pull request #241 from borod108/bugs/1571291snapshot_removal
    
    Fix refreshing targeted vm after snapshot removal
    (cherry picked from commit 233273d4a229a5580ef8615e4c725d09a87dc733)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1581287
```

**Pull request merged**